src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -293,6 +293,8 @@ std::unique_ptr<WriteBufferFromFileBase> S3ObjectStorage::writeObject( /// NOLIN
 ObjectStorageIteratorPtr S3ObjectStorage::iterate(const std::string & path_prefix, size_t max_keys) const
 {
     auto settings_ptr = s3_settings.get();
+    if (!max_keys)
```

Comment:
Here was the error-prone code: the user of the iterator had to pass something like 1000 explicitly. Now, it is not needed.